File: expense-tracker/backend/src/main/java/com/expensetracker/service/CategoryService.java
package com.expensetracker.service;

import com.expensetracker.dto.request.CategoryRequest;
import com.expensetracker.dto.response.CategoryResponse;

import java.util.List;

public interface CategoryService {
    CategoryResponse create(Long userId, CategoryRequest request);
    List<CategoryResponse> getAll(Long userId);
    CategoryResponse update(Long userId, Long id, CategoryRequest request);
    void delete(Long userId, Long id);
}

File: expense-tracker/frontend/src/services/categoryService.js
import api from './api';

const categoryService = {
  async getAll() {
    const { data } = await api.get('/categories');
    return data;
  },

  async create(payload) {
    const { data } = await api.post('/categories', payload);
    return data;
  },

  async update(id, payload) {
    const { data } = await api.put(`/categories/${id}`, payload);
    return data;
  },

  async remove(id) {
    await api.delete(`/categories/${id}`);
  },
};

export default categoryService;
